azure-mgmt-sql/src/main/java/com/microsoft/azure/management/sql/implementation/SqlServerImpl.java

```diff
@@ -218,19 +226,24 @@ public SqlServerImpl withAdministratorPassword(String administratorLoginPassword
     }
 
     @Override
-    public SqlServerImpl withVersion(ServerVersion version) {
-        this.inner().withVersion(version);
-        return this;
-    }
-
-    @Override
     public SqlServerManager manager() {
         return this.myManager;
     }
 
     @Override
     public SqlServerImpl withNewDatabase(String databaseName) {
-        this.withDatabaseInElasticPool(databaseName, null);
+        if (this.databaseCreatableMap.get(databaseName) != null) {
```

Comment:
assuming this is a Map<K,V>, you don't need to check for the existence of the element in the map to be able to remove it. Just calling remove is enough.